### ISSUE TITLE
[general need] FSL is no longer working in mia

### ISSUE THREAD

**Issue by servoz** [populse/populse_mia#265]
It is not known since how long.
First investigation indicate that with all populse on master:
  - [capsul at 618f3fa0 commit and populse_mia at aeb88cb4 commit => it's working](https://github.com/populse/populse_mia/issues/263#issuecomment-1103986840)
  - capsul at 618f3fa0 commit and populse_mia at d640d339 => it's not working
exception:
 ```
 File "<string>", line 1, in <module>
  File "/casa/host/src/capsul/master/capsul/process/process.py", line 731, in run_from_commandline
    result = ce.study_config.run(process, configuration_dict=configuration)
  File "/casa/host/src/capsul/master/capsul/study_config/study_config.py", line 429, in run
    configuration_dict=configuration_dict)
  File "/casa/host/src/capsul/master/capsul/study_config/run.py", line 165, in run_process
    returncode = process_instance._run_process()
  File "/casa/home/Git_projects/populse_mia/python/populse_mia/user_interface/pipeline_manager/process_mia.py", line 518, in _run_process
    self.run_process_mia()
  File "/casa/home/Git_projects/mia_processes/python/mia_processes/bricks/preprocess/fsl/processes.py", line 230, in run_process_mia
    return self.process.run(configuration_dict={})
  File "/casa/host/src/capsul/master/capsul/process/process.py", line 310, in run
    return self.__call__(**kwargs)
  File "/casa/host/src/capsul/master/capsul/process/process.py", line 302, in __call__
    returncode = self.get_study_config().run(self, **kwargs)
  File "/casa/host/src/capsul/master/capsul/study_config/study_config.py", line 429, in run
    configuration_dict=configuration_dict)
  File "/casa/host/src/capsul/master/capsul/study_config/run.py", line 165, in run_process
    returncode = process_instance._run_process()
  File "/casa/host/src/capsul/master/capsul/process/process.py", line 1877, in _run_process
    results = self._nipype_interface.run()
  File "/casa/home/.local/lib/python3.6/site-packages/nipype-1.7.1-py3.6.egg/nipype/interfaces/base/core.py", line 398, in run
    runtime = self._run_interface(runtime)
  File "/casa/home/.local/lib/python3.6/site-packages/nipype-1.7.1-py3.6.egg/nipype/interfaces/base/core.py", line 756, in _run_interface
    % (executable_name, runtime.hostname)
OSError: No command "fslmaths" found on host irmage-rico. Please check that the corresponding package is installed.

```

Same result using nipype 1.7.0 (not tested before)

Quick and dirty solution while waiting for a fix:
Add in .bashrc:
```
FSLDIR=<Where is installed fsl>
. ${FSLDIR}/etc/fslconf/fsl.sh
PATH=${FSLDIR}/bin:${PATH}
export FSLDIR PATH
```

**Comment by LStruber** [comment]
In software_properties.py, if I change line 392:
`m['directory'] = os.path.dirname(fsl_config)`
by:
`m['directory'] = os.path.dirname('/home/lucas/apps/fsl/bin')`

It works. Of course it should be done cleanly (by removing 'fslconf' at the end of fsl_config directory and adding 'bin'). But as you are the one that works a lot on synchronization between Capsul config and MIA preferences, you may potential implication/alter effects that I don't see. Any ideas ?

**Comment by LStruber** [comment]
> by removing 'fslconf' at the end of fsl_config directory and adding 'bin'

Adding bin is not necessary. The following lines are working:

```
m['directory'] = os.path.split(
                os.path.split(
                    os.path.dirname(fsl_config))[0])[0]
```

I'm waiting for your opinion before pushing

**Comment by servoz** [comment]
Oh great !!
You found the issue exactly !!!

It looks like a (big) typo :-).

I would prefer to replace
l392 `m['directory'] = os.path.dirname(fsl_config)`
with
```
m['directory'] = os.path.dirname(
                                 os.path.dirname(
                                     os.path.dirname(fsl_config)))
```

but indeed
```
m['directory'] = os.path.split(
                os.path.split(
                    os.path.dirname(fsl_config))[0])[0]

```
do also the job !

I will leave to you the pleasure to push the change and close this ticket.
Cool!

**Comment by LStruber** [comment]
closed with 73f121707d25b3f2ee3d28cb57981313cd6ea6cd
